### PR TITLE
refactor(ava/data): migrate and improve statistics

### PR DESCRIPTION
**Pull request by pddpd**
Add statistical methods used internally by ava.
In order to CR easily, I split commits into multiple PRs. This PR is merged on the premise that [feat(ava/data): add utils](https://github.com/antvis/AVA/pull/479) is merged.

### CODE REVIEW

File: packages/ava/src/data/statistics/index.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
+/**
+ * Statistical methods used internally by ava.
+ */
+
+import { assert } from '../utils';
```
Comment (pddpd) [inline packages/ava/src/data/statistics/index.ts:5]: dependency: [feat(ava/data): add utils](https://github.com/antvis/AVA/pull/479)

File: packages/ava/src/data/statistics/index.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
+/**
```
Comment (neoddish) [inline packages/ava/src/data/statistics/index.ts:1]:
```suggestion
/*
```

this is a multi-line comment for the file...not jsdoc

File: packages/ava/src/data/statistics/index.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    value,
+    'sum',
+    value.reduce((prev, current) => current + prev, 0)
+  );
+}
+
+/**
+ * Calculate the mean of the array.
+ * @param value - The array to process
+
```
Comment (neoddish) [inline packages/ava/src/data/statistics/index.ts:92]: ?

File: packages/ava/src/data/statistics/index.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+      min = value[i];
+    }
+  }
+  return idx;
+}
+
+/**
+ * Calculate the minimum index of the array.
+ * @param value - The array to process
+ */
```
Comment (neoddish) [inline packages/ava/src/data/statistics/index.ts:36]:
```suggestion
/**
 * Calculate the minimum index of the array.
 * ---
 * 得到数组中最小值的 index
 *
 * @param value - The array to process
 */
```

could add some Chinese comments for some jsdocs

File: packages/ava/src/data/statistics/index.ts [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+/**
+ * Calculate the minimum of the array.
+ * @param value - The array to process
+ */
+export function min(value: number[]): number {
+  const cachedValue = cache.get<number>(value, 'min');
+  if (cachedValue !== undefined) {
+    return cachedValue;
+  }
+  return cache.set(value, 'min', Math.min(...value));
```
Comment (BBSQQ) [inline packages/ava/src/data/statistics/index.ts:18]: Do we need declare constants for key?

Comment (pddpd) [reply packages/ava/src/data/statistics/index.ts:18]: It doesn't matter. Function names cannot use constants, and these names are only used here.

File: packages/ava/src/data/statistics/index.ts (hunk 2 above)
Comment (pddpd) [reply packages/ava/src/data/statistics/index.ts]: done

**Review by neoddish — APPROVED** [review]
lgtm

**Pull request merged**